libs/domains/variables/feature/src/lib/create-update-variable-modal/create-update-variable-modal.tsx

```diff
@@ -0,0 +1,366 @@
+import { type APIVariableScopeEnum, type APIVariableTypeEnum, type VariableResponse } from 'qovery-typescript-axios'
+import { useState } from 'react'
+import { Controller, FormProvider, useForm } from 'react-hook-form'
+import { match } from 'ts-pattern'
+import {
+  Icon,
+  InputSelect,
+  InputText,
+  InputTextArea,
+  InputToggle,
+  ModalCrud,
+  Tooltip,
+  useModal,
+} from '@qovery/shared/ui'
+import {
+  computeAvailableScope,
+  environmentVariableFile,
+  generateScopeLabel,
+  getEnvironmentVariableFileMountPath,
+} from '@qovery/shared/util-js'
+import { useCreateVariableAlias } from '../hooks/use-create-variable-alias/use-create-variable-alias'
+import { useCreateVariableOverride } from '../hooks/use-create-variable-override/use-create-variable-override'
+import { useCreateVariable } from '../hooks/use-create-variable/use-create-variable'
+import { useEditVariable } from '../hooks/use-edit-variable/use-edit-variable'
+
+export type CreateUpdateVariableModalProps = {
+  parentId: string
+  scope: Exclude<keyof typeof APIVariableScopeEnum, 'BUILT_IN'>
+  closeModal: () => void
+  onSubmit?: (variable?: VariableResponse | void) => void
+  variable?: VariableResponse
+  mode: 'CREATE' | 'UPDATE'
+  type: keyof typeof APIVariableTypeEnum
+  isFile?: boolean
+}
+
+export function CreateUpdateVariableModal({
+  parentId,
+  scope,
+  closeModal,
+  onSubmit: _onSubmit,
```

Comment:
Should be the inverse, same you have done on the `VariablesActionToolbar` component.
_onSubmit is externe

Reply:
https://github.com/Qovery/console/pull/1318/commits/b9aa846a31eb056e479de8bbc75c408d2154dba7